facebook_connector.py

```diff
@@ -0,0 +1,40 @@
+from fbchat_archive_parser.parser import MessageHtmlParser
+import fnmatch
+import os
+import pandas as pd
+
+
+def initialize(dump_location='.'):
+    global messages_file
+    fb_message_filename = 'messages.htm'
+    messages_file = os.path.join(dump_location, fb_message_filename)
+    if not os.path.isfile(messages_file):
+        print("Please place the message.htm file at {}".format(
+            os.path.abspath(messages_file)))
+        return
+
+
+def get_cleaned_fully_merged_messages():
```

Comment:
I dug through this data a bit more and there's a key issue that I noticed.  In your dataframe, if the message is `is_from_me` then you have the `full_name` set to me.  This means you never have a record of who I've messaged which is a problem, instead it looks like I just messaged myself.  You can see this if you run `fully_merged_messages_df[fully_merged_messages_df.is_from_me]` in the notebook.

However, this information is available in the thread.partcipants section so you just have to merge it in.  One way to do this is to alter the messages list to actually be a list of dictionaries of `{message: <text>, 'is_from_me': <true/false>, 'full_name': <recipient/sender>}`.  You can do this by applying a for loop on the participant list for every single message that you've sent to indicate that it was indeed sent to each person.  Then for the dataframe you can create it directly from that list via: http://pandas.pydata.org/pandas-docs/stable/dsintro.html#from-a-list-of-dicts

Also, no need to worry about the 80 char limit :) I should've mentioned that I've been somewhat ignoring that part of pep8 myself...